tensorflow/lite/micro/kernels/concatenation.cc

```diff
@@ -155,62 +176,53 @@ TfLiteStatus ConcatenationPrepare(TfLiteContext* context, TfLiteNode* node) {
           RuntimeShape::kMaxSmallSize, num_dimensions);
       return kTfLiteError;
     }
+
+    if (input_type == kTfLiteInt8) {
+      // Make sure there is no re-scaling needed for Int8 quantized kernel. This
+      // is a restriction we introduced to Int8 kernels.
+      TF_LITE_ENSURE_EQ(context, static_cast<double>(input->params.scale),
+                        static_cast<double>(output_tensor->params.scale));
+      TF_LITE_ENSURE_EQ(context, input->params.zero_point,
+                        output_tensor->params.zero_point);
+    } else if (input_type == kTfLiteInt16) {
+      // Make sure that all Int16 inputs have a null zero-point.
+      TF_LITE_ENSURE_EQ(context, input->params.zero_point, 0);
```

Comment:
Is there a reason why the scale is not checked in the int16 case? AFAICS the (at least the reference) concatenation kernel does not take into account any scaling/quantization at all.

https://github.com/tensorflow/tflite-micro/blob/b2f27186d243d6afc2324c43b3bf9a9f681be474/tensorflow/lite/kernels/internal/reference/concatenation.h#L30

Reply:
I agree with your conclusion, however this is a copy-paste of the TfLite code.

Reply:
Yes, also is there are reason why it has to be zero? Seems that concatenation in general shouldn't care about quantization parameters, as long as they match in and out.

Reply:
@tinskip This same check is performed by the LiteRT (TfLite) reference implementation.